Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/SettingsFile.java

```diff
@@ -23,6 +23,33 @@
 import java.util.TreeSet;
 
 /**
+ * A HashMap<String, SettingSection> that constructs a new SettingSection instead of returning null
+ * when getting a key not already in the map
+ */
+final class SettingsSectionMap extends HashMap<String, SettingSection>
+{
+	@Override
+	public SettingSection get(Object key)
+	{
+		if (!(key instanceof String))
+		{
+			return null;
+		}
+		String stringKey = (String)key;
+		if (!super.containsKey(stringKey))
+		{
+			SettingSection section = new SettingSection(stringKey);
+			super.put(stringKey, section);
+			return section;
+		}
+		else
```

Comment:
The `else` keyword and its braces can be omitted, since the above `if` has a return in it.

Reply:
I felt the if() { return} else {return} was more obvious - but I tweaked it to remove the early return anyway.

Reply:
The general convention in the codebase is to just early return and omit the braces on the else if the related if statement also contains a return like so (similar to [LLVM's style guideline](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return)):

```java
if (!super.containsKey(stringKey))
{
    SettingSection section = new SettingSection(stringKey);
    super.put(stringKey, section);
    return section;
}

return super.get(key);
```